### PR TITLE
add cohortID to logEvent

### PR DESCRIPTION
**Pull request by camdencheek**
This commit adds an optional cohortID to each logEvent call. This allows
us to identify users by a cohortID for A/B testing. It's expected the
cohortID is a date in the form of YYYY-MM-DD.

### CODE REVIEW

File: cmd/frontend/graphqlbackend/schema.graphql [hunk 1]
```diff
@@ -417,6 +417,11 @@ type Mutation {
         The source of the event.
         """
         source: EventSource!
+        """
+              An optional cohort ID to identify the user as part of a specific A/B test.
+        The cohort ID is expected to be a date in the form YYYY-MM-DD
```
Comment (limitedmage) [inline cmd/frontend/graphqlbackend/schema.graphql:422]:
```suggestion
        An optional cohort ID to identify the user as part of a specific A/B test.
        The cohort ID is expected to be a date in the form YYYY-MM-DD
```

**Review by limitedmage — COMMENTED** [review]
How/when is the cohort ID generated?

**Comment by camdencheek** [comment]
> How/when is the cohort ID generated?

For this week's A/B test, cohortID should be added to all new users. The cohortID for the current experiment would be `2020-06-21`, representing the test that will be running next week. https://sourcegraph.slack.com/archives/CHEKCRWKV/p1623760995329900

**Review by limitedmage — APPROVED** [review]

File: cmd/frontend/graphqlbackend/schema.graphql [hunk 2]
```diff
@@ -418,6 +418,11 @@ type Mutation {
         """
         source: EventSource!
         """
+        An optional cohort ID to identify the user as part of a specific A/B test.
+        The cohort ID is expected to be a date in the form YYYY-MM-DD
+        """
+        cohortID: String
```
Comment (limitedmage) [inline cmd/frontend/graphqlbackend/schema.graphql:424]: Should we also add this to `logUserEvent`?

Comment (camdencheek) [reply cmd/frontend/graphqlbackend/schema.graphql:424]: Oh. Good question. What's the difference between `logEvent` and `logUserEvent`?

Comment (limitedmage) [reply cmd/frontend/graphqlbackend/schema.graphql:424]: Looks like every use of `logUserEvent` has a corresponding `logEvent` so I don't think we should add it. `logUserEvent` is deprecated.

**Pull request merged**